internal/app/go-filecoin/internal/submodule/messaging_submodule.go

```diff
@@ -38,14 +39,22 @@ func NewMessagingSubmodule(ctx context.Context, config messagingConfig, repo mes
 	msgPool := message.NewPool(repo.Config().Mpool, consensus.NewIngestionValidator(chain.State, repo.Config().Mpool))
 	inbox := message.NewInbox(msgPool, message.InboxMaxAgeTipsets, chain.ChainReader, chain.MessageStore)
 
+	// setup messaging topic
+	topic, err := network.Fsub.Join(net.MessageTopic(network.NetworkName))
+	if err != nil {
+		return MessagingSubmodule{}, err
+	}
```

Comment:
I'm a bit confused about the difference between this `Join` and `Topic.Subscribe()` calls below. I've been code diving to try to reconcile this with the code I see in libp2p (on master) but haven't 100% figured it out.

Subscribing (having any network side-effect) at module construction is undesirable. It looks like in libp2p, this `Join()` is a no-op (but also with no return value?). If that's an assumption we're making in order to not have side-effects here, we should document as such. But with GossipSub, `Join` does do a bunch, so we're not getting any abstraction here.

If constructing a Topic object involves interacting with libp2p in a way that has side effects, I think we're better representing a topic as a string until we actually want to subscribe to it.

Reply:
I want to move away from the deprecated endpoints so I'm keeping Topics.  However your point on side effects during construction is well taken.  I'll construct both the topic and subscription during start.